WordPress/gutenberg
# Remove snapshots from tests for LinkControl

## Description
In https://github.com/WordPress/gutenberg/pull/31464 @youknowriad said:

> For the future, it would be cool if we try to limit the use of snapshots in tests (specially big ones). These tests are harder to debug and change to often when implementation details do.

I completely agree. I'm not generally a fan of snapshots and find them to all little value to the majority of tests. To this end I've removed the snapshots from the tests for LinkControl.

Note how having removed said snapshots the utility of the tests has not diminished. This is because the underlying tests were already written in a manner such that they made assertions directly.

That being said, in a follow up I'm going to utilise React Testing Library to convert some assertions over to text matches to avoid relying on `classNames` which - after all - are not a user-perceiveable API.

## How has this been tested?

* Check tests for any remaining usage of "snapshots".
* Run tests - check they are green.

## Screenshots 

## Types of changes





## Checklist:
